### PR TITLE
Update Status Bar to Match Theme Colors Dynamically

### LINKED ISSUE #2710: Bug: Status Bar Background Not Adhering to Theme
**Summary:**

The status bar background is not correctly set according to the theme, causing a visual mismatch between the app’s UI and the status bar.

**Expected behaviour:**

The expected behavior is for the status bar background to follow the app's theme (e.g., colors, transparency), ensuring consistency throughout the UI

**Screenshots:**

| Light Theme | Dark Theme |
|---|---|
| WhatsApp Image 2024-09-20 at 07 18 27_ea2a1732 | WhatsApp Image 2024-09-20 at 07 18 28_e1b27c20
 |

### PR DESCRIPTION
**Pull request by Saifuddin53**
Fixes #2710

Please Add Screenshots If there are any UI changes.

Light theme: 

| Before | After |
|---|---|
| light | light

Dark theme: 

| Before | After |
|---|---|
| dark | dark

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt [hunk 1]
```diff
@@ -50,6 +55,15 @@ fun MifosMobileTheme(
...
     }
 
+    val view = LocalView.current
+    if (!view.isInEditMode) {
+        SideEffect {
+            val window = (view.context as Activity).window
+            window.statusBarColor = colors.primary.toArgb()
+            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !useDarkTheme
+        }
+    }
```
Comment (niyajali) [inline core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt]: Use `enableEdgeToEdge()` API, and set status bar color to **Transperant** in both light and dark theme.

**Review by niyajali — CHANGES_REQUESTED** [review]

File: core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt [hunk 2]
```diff
@@ -50,6 +56,15 @@ fun MifosMobileTheme(
         else -> LightThemeColors
     }
 
+    val view = LocalView.current
+    if (!view.isInEditMode) {
+        SideEffect {
```
Comment (niyajali) [inline core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt]: we no need to configuring it manually this should be provided by EnableEdgeToEdge Api, you could configure theme for night & dark mode like below

update `splash.xml`
```
<resources xmlns:tools="http://schemas.android.com/tools">

    <!-- Allows us to override night specific attributes in the
     values-night folder. -->
    <style name="NightAdjusted.Theme" parent="android:Theme.Material.Light.NoActionBar">
        <item name="android:statusBarColor">@color/status_bar</item>
    </style>

    <!-- The final theme we use -->
    <style name="Theme.Nia" parent="NightAdjusted.Theme" />

    <style name="NightAdjusted.Theme.Splash" parent="Theme.SplashScreen">
        <item name="android:windowLightNavigationBar" tools:targetApi="27">true</item>
    </style>

    <style name="Theme.Mifos.Splash" parent="NightAdjusted.Theme.Splash">
        <item name="windowSplashScreenAnimatedIcon">@drawable/splash_icon</item>
        <item name="postSplashScreenTheme">@style/Theme.Nia</item>
    </style>

</resources>
```

create values-night folder and create `theme.xml, colors.xml` file
values-night/theme.xml would be look like this
```
<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:tools="http://schemas.android.com/tools">

    <style name="NightAdjusted.Theme" parent="android:Theme.Material.NoActionBar">
        <item name="android:statusBarColor">@color/status_bar</item>
    </style>

    <style name="NightAdjusted.Theme.Splash" parent="Theme.SplashScreen">
        <item name="android:windowLightNavigationBar" tools:targetApi="27">false</item>
    </style>

</resources>
```
`color.xml`
```
<color name="status_bar">#FFFEFBFF</color> // light/color.xml
<color name="status_bar">#FF1B1B1F</color> // dark/color.xml
```
`HomeActivity.kt`

```
       setContent {
            val darkTheme = isSystemInDarkTheme()

            // Update the dark content of the system bars to match the theme
            DisposableEffect(darkTheme) {
                enableEdgeToEdge(
                    statusBarStyle = SystemBarStyle.auto(
                        Color.TRANSPARENT,
                        Color.TRANSPARENT,
                    ) { darkTheme },
                    navigationBarStyle = SystemBarStyle.auto(
                        lightScrim,
                        darkScrim,
                    ) { darkTheme },
                )

                onDispose {}
            }
     }
```

you may need to configure TopAppBar colors as well, update DarkThemeColors
`surface = BackgroundDark` 

you may need to do some small tweaks to match these StatusBar, NavigationBar and Surface colors to same.

**Review by niyajali — CHANGES_REQUESTED** [review]

File: core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt (hunk 2 above)
Comment (Saifuddin53) [reply core/designsystem/src/main/kotlin/org/mifos/mobile/core/designsystem/theme/Theme.kt]: @niyajali I have updated the code and screenshots can you please recheck it

**Comment by therajanmaurya** [comment]
is it ready for final review @niyajali

**Comment by niyajali** [comment]
@therajanmaurya looks good to me..and it's ready for the final review

**Comment by Saifuddin53** [comment]
@therajanmaurya can we merge this PR?

**Pull request closed without merging**